detect_secrets/core/scan.py

```diff
@@ -206,6 +206,9 @@ def scan_for_allowlisted_secrets_in_file(filename: str) -> Generator[PotentialSe
     except IOError:
         log.warning(f'Unable to open file: {filename}')
         return
+    except ValueError:
+        log.warning(f'Unable to scan file: {filename}. Please ignore if file is empty.')
+        return
```

Comment:
I would suggest checking `if lines:` above instead - that way we'll be specifically addressing the issue (as opposed to any `ValueError` exception) and we won't need to execute the line `yield from _scan_for_allowlisted_secrets_in_lines(enumerate(lines, start=1), filename)` at all